test/testdata/lsp/completion/enum.rb

```diff
@@ -0,0 +1,5 @@
+# typed: true
+class Test
+  enum # error: does not exist
+  #   ^ apply-completion [A] item: 0
+end
```

Comment:
Could we get a couple tests:

- What happens when there is a local variable in scope with the name `enum`?
- What about when there is a method in scope on `self` with the name `enum`?

Reply:
Another test: can we make sure that

```ruby
x.enum
```

where `x` is some type that has an `enum` method does not produce any results?

Reply:
Added some more tests to cover this!